src/Formatter.js

```diff
@@ -65,6 +65,17 @@ class Formatter {
      */
     format(result) {
     }
+    /**
+     * Provide the information for the formatter as an object and
+     * return the formatted string that contains a lot of information
+     * according to the formatter's needs.
+     *
+     * @abstract
+     * @param {Object} [options] the options that needs for formatter
```

Comment:
"Information that the method can use to format the output"

Then, add documentation for each property
@param {string} [options.errors] total number of error results found in this run
etc.